scipy/optimize/_differentialevolution.py

```diff
@@ -823,6 +957,49 @@ def __del__(self):
         self._mapwrapper.close()
         self._mapwrapper.terminate()
 
+    def _accept_trial(self, energy_trial, feasible_trial, cv_trial,
+                      energy_orig, feasible_orig, cv_orig):
+        """
+        Trial is accepted if:
+        * it satisfies all constraints and provides a lower or equal objective
+          function value, while both the compared solutions are feasible
+        - or -
+        * it is feasible while Xi, G is infeasible,
+        - or -
+        * it is infeasible, but provides a lower or equal constraint violation
+          for all constraint functions.
+
+        Parameters
+        ----------
+        energy_trial : float
+            Energy of the trial solution
+        feasible_trial : float
+            Feasibility of trial solution
+        cv_trial : array-like
+            Excess constraint violation for the trial solution
+        energy_orig : float
+            Energy of the original solution
+        feasible_orig : float
+            Feasibility of original solution
+        cv_orig : array-like
+            Excess constraint violation for the original solution
+
+        Returns
+        -------
+        accepted : bool
+
+        """
+        if feasible_orig and feasible_trial:
+            return energy_trial < energy_orig
```

Comment:
Technically in Lampinen (Equation 9 and in English below) this is `<=`

Reply:
Yes. `<` has been employed in `scipy` up until now. However, going back to many of the papers that I've got on DE (including Lampinen) they all use `<=`. This seems to be an error in the scipy implementation since it was first included. However, a pragmatic assessment of it's effect would indicate that it's unlikely to have a big effect.

Reply:
This will be a backwards incompatible change, but it's technically fixing a bug. Anyone relying on tests producing specific numeric output is going to be disappointed.